src/robotFactory.js

```diff
@@ -1,10 +1,101 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion
+  ) {
+    this.name = name;
+    this.weight = weight;
+    this.weight = weight;
 
```

Comment:
looks like same data
```suggestion
    this.weight = weight;

```